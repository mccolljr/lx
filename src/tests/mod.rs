use std::io::Read;
use std::process::{
    Command,
    Stdio,
};

#[test]
fn test_snapshots() {
    glob!("testdata/*.lx", |path| {
        println!("testing {:?}", path);
        let mut child = Command::new("cargo")
            .arg("run")
            .arg(path)
            .stdout(Stdio::piped())
            .spawn()
            .expect("failed to spawn test");
        let exit_status = child.wait().expect("failed to run test");
        let mut output_str = String::new();
        child
            .stdout
            .expect("no stdout for test")
            .read_to_string(&mut output_str)
            .expect("can't read test output");

        if path.to_str().unwrap().ends_with("_error.lx") {
            if exit_status.success() {
                panic!(
                    "expected failure, got success: {:?}\n{}",
                    path, output_str
                );
            }
        } else {
            if !exit_status.success() {
                panic!(
                    "expected success, got failure: {:?}\n{}",
                    path, output_str
                );
            }
        }
        assert_snapshot!(output_str);
    });
}
