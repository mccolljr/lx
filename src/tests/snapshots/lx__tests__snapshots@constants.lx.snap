---
source: src/tests/mod.rs
expression: output_str
input_file: src/tests/testdata/constants.lx
---
1
1.5
0.0000000000015
def
abc

abc
    def
        ghi
            jkl

true
false
null
OK!

